app/models/ReferenceDataPayload.scala

```diff
@@ -40,8 +45,28 @@ class ReferenceDataPayload(data: JsObject) {
     )
 }
 
-object ReferenceDataPayload {
+object ReferenceDataListsPayload {
+  def apply(v1: JsObject): ReferenceDataListsPayload = new ReferenceDataListsPayload(v1)
+}
+
+class CustomsOfficeListsPayload(data: JsObject) extends ReferenceDataPayload {
+
+  override lazy val messageInformation: MessageInformation =
+    (for {
+      JsString(messageId) <- (data \ "messageInformation" \ "messageID").validate[JsString]
+      snapshotDate        <- (data \ "messageInformation" \ "snapshotDate").validate[LocalDate]
+    } yield MessageInformation(messageId, snapshotDate)).getOrElse(throw new Exception("WTF"))
```

Comment:
Maybe not the most ideal exception xD

Reply:
Can you make this change please? This was a bit of debugging